click/testing.py

```diff
@@ -292,14 +294,19 @@ def invoke(self, cli, args=None, input=None, env=None,
 
                 exit_code = e.code
                 if not isinstance(exit_code, int):
+                    if exit_code is None:
```

Comment:
You need to do this outside of the outer if, because otherwise we will print `"0"`.